src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/lib/__tests__/get_matched_indices.test.js

```diff
@@ -1,172 +1,96 @@
-      partialMatchedIndices: [{ name: 'kibana' }, { name: 'kibana2' }],
-      visibleIndices: [{ name: 'kibana' }, { name: 'kibana2' }]
+    it('should return unfiltered', () => {
+      const query = 'ki*';
+      const { exactMatchedIndices } = getMatchedIndices(indices, indices, query, true);
+      expect(exactMatchedIndices).toEqual([
+        { name: 'kibana' },
+        { name: '.kibana' },
+      ]);
     });
   });
 
-  it('should return all indices as visible if there are no partial or exact matches', () => {
-    const indices = [
-      { name: 'kibana' },
-      { name: 'es' },
-      { name: '.kibana' }
-    ];
-
-    const query = 'fo';
-    const matchedIndices = [];
-
-    const result = getMatchedIndices(indices, matchedIndices, query, false);
-
-    expect(result).toEqual({
-      allIndices: [{ name: 'kibana' }, { name: 'es' }],
-      exactMatchedIndices: [],
-      partialMatchedIndices: [],
-      visibleIndices: [{ name: 'kibana' }, { name: 'es' }]
+  describe('partialMatchedIndices', () => {
+    it('should return filtered', () => {
+      const query = 'ki*';
+      const partialIndices = indices.slice(1);
+      const { partialMatchedIndices } = getMatchedIndices(indices, partialIndices, query, false);
+      expect(partialMatchedIndices).toEqual([
+        { name: 'es' },
+        { name: 'logstash' },
+        { name: 'packetbeat' },
+        { name: 'metricbeat' },
+      ]);
     });
-  });
 
-  it('should support showing system indices', () => {
-    const indices = [
-      { name: 'kibana' },
-      { name: 'es' },
-      { name: '.kibana' }
-    ];
-
-    const query = 'ki';
-    const matchedIndices = [
-      { name: '.kibana' },
-      { name: 'kibana' },
-    ];
-
-    const result = getMatchedIndices(indices, matchedIndices, query, true);
-
-    expect(result).toEqual({
-      allIndices: [{ name: 'kibana' }, { name: 'es' }, { name: '.kibana' }],
-      exactMatchedIndices: [],
-      partialMatchedIndices: [{ name: '.kibana' }, { name: 'kibana' }],
-      visibleIndices: [{ name: '.kibana' }, { name: 'kibana' }]
+    it('should return unfiltered', () => {
+      const query = 'ki*';
+      const partialIndices = indices.slice(1);
+      const { partialMatchedIndices } = getMatchedIndices(indices, partialIndices, query, true);
+      expect(partialMatchedIndices).toEqual(partialIndices);
     });
   });
 
-  it('should only return the max number of indices', () => {
-    const indices = [
-      { name: 'kibana' },
-      { name: 'es' },
-      { name: '.kibana' },
-      { name: 'monitor' },
-      { name: '.monitor' },
-      { name: 'metricbeat' },
-    ];
-
-    const query = '';
-    const matchedIndices = [
-      { name: 'kibana' },
-      { name: 'es' },
-      { name: '.kibana' },
-      { name: 'monitor' },
-      { name: '.monitor' },
-      { name: 'metricbeat' },
-    ];
-
-    const result = getMatchedIndices(indices, matchedIndices, query, true);
+  describe('visibleIndices', () => {
```

Comment:
Re our convo, I think these test would be easier to follow if we reverse the order of the "filtered" and "not filtered" assertions, so that the reader understands the baseline behavior first, and then can compare that with the "filtered" test to see how changing the last argument to `false` affects the output.

I also think being more verbose with the description would help make these tests clearer, e.g. by changing "should return unfiltered" to "returns all indices" and "should return filtered" to "excludes system indices".

Reply:
Agreed!